add-on/src/lib/telemetry.js

```diff
@@ -0,0 +1,60 @@
+
+import { MetricsProvider } from '@ipfs-shipyard/ignite-metrics/vanilla'
+
+let metricsProvider = null
+export function getMetricsProviderInstance () {
+  if (metricsProvider != null) {
+    return metricsProvider
+  }
+  metricsProvider = new MetricsProvider({ appKey: '393f72eb264c28a1b59973da1e0a3938d60dc38a', autoTrack: false })
+
+  return metricsProvider
+}
+
+/**
+ * @param {ReturnType<import('./state').initState>['options']} state
+ * @returns {string[]}
+ */
+function mapStateToConsent (stateOptions) {
+  const obj = {
+    minimal: stateOptions?.telemetryGroupMinimal || false,
+    marketing: stateOptions?.telemetryGroupMarketing || false,
+    performance: stateOptions?.telemetryGroupPerformance || false,
+    tracking: stateOptions?.telemetryGroupTracking || false
+  }
+
+  const enabledConsentGroups = Object.keys(obj).filter(key => obj[key] === true)
+  console.log('enabledConsentGroups: ', enabledConsentGroups)
+  return enabledConsentGroups
+}
+function logConsent () {
+  console.log('checkConsent(\'minimal\'): ', getMetricsProviderInstance().checkConsent('minimal'))
+  console.log('checkConsent(\'marketing\'): ', getMetricsProviderInstance().checkConsent('marketing'))
+  console.log('checkConsent(\'performance\'): ', getMetricsProviderInstance().checkConsent('performance'))
+  console.log('checkConsent(\'tracking\'): ', getMetricsProviderInstance().checkConsent('tracking'))
```

Comment:
Remove all `console.log` calls, use scoped `debug` logger instead – see `src/lib/on-installed.js`, we want something similar:

```js
import debug from 'debug'
const log = debug('ipfs-companion:telemetry')
log(`checkConsent(foo): ${bar}`)
```
```